DigitalLearningSolutions.Data/Services/ImageResizeService.cs

```diff
@@ -67,8 +88,25 @@ private Image CropImageToCentredSquare(Image image)
 
         private Image ResizeSquareImage(Image image, int sideLengthPx)
         {
-            var destRect = new Rectangle(0, 0, sideLengthPx, sideLengthPx);
-            var returnImage = new Bitmap(sideLengthPx, sideLengthPx);
+            return ResizeImageToDimensions(image, sideLengthPx, sideLengthPx);
+        }
+
+        private Image ResizeImageByMaxSideLength(Image image, int maxSideLengthPx)
+        {
+            var ratioX = Math.Min((float)maxSideLengthPx / (float)image.Width, 1);
+            var ratioY = Math.Min((float)maxSideLengthPx / (float)image.Height, 1);
+            var ratio = Math.Min(ratioX, ratioY);
```

Comment:
Is it worth just taking Max on image width and image height and then calculating ratio based on that?
```
var longerSideOfImageInPx = Max(width, height)
// No need to resize if image is smaller than the possible max
var ratio = Min(maxSideLengthPx / longerSideOfImageInPx, 1)
```

(might be more intuitive than ratioX/ratioY and then ratio based on the ratios)